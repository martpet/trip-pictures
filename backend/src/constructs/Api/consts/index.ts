export * from './authorizationHeader';
export * from './lambdaEdgeViewerEvent';
