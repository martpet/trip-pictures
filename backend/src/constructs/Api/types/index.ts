export * from './Cookies';
export * from './EnvVars';
export * from './LambdaEdgeOriginRequestHandler';
export * from './LambdaEdgeResponse';
export * from './OauthCookieProps';
export * from './OauthTokens';
export * from './RouteHeaders';
