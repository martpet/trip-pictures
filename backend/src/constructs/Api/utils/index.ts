export * from './cookieName';
export * from './ddbHelpers';
export * from './edgeErrorResponse';
export * from './errorResponse';
export * from './getIdTokenPayload';
export * from './parseEventCookies';
export * from './parseLambdaEdgeEventCookies';
export * from './testHelpers';
