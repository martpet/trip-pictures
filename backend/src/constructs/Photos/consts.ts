export const IS_PHOTOS_TABLE_ITEM_CREATED_S3_TAG = 'isTableItemCreated';
export const STREAM_HANDLER_PHOTO_BUCKET_ENV_KEY = 'photoBucket';
