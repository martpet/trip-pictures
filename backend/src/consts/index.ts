export * from '~/../../shared/consts';
//
export * from './appConsts';
export * from './appEnvs';
export * from './authPaths';
export * from './tablesOptions';
