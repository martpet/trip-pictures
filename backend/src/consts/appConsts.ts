import { localhostPort } from '~/../../shared/consts';

export const authSubdomain = 'auth';
export const localhostUrl = `http://localhost:${localhostPort}`;
export const healthCheckAlarmEmails = ['martin_petrov@me.com'];
export const prodAccountId = '766373560006';
export const stagingAccountId = '204115048155';
export const rootHostedZoneId = 'Z0207937145S41OBP9QS3';
export const stagingHostedZoneId = 'Z01791432G9JNVDM92SAJ';
export const devHostedZoneId = 'Z01228832S4EGTRIOWEIT';
export const devAccountServiceRoleArn = 'arn:aws:iam::020463219829:role/DevAccountServiceRole';
export const googleClientSecretParamName = '/oauth/google/client-secret';
export const googleClientIdProd = '276806659709-nbtr116utpr7ak4j43gfq7sfi0frtcra.apps.googleusercontent.com';
export const googleClientIdStaging = '276806659709-du1tv5u4juububcu61o1hn5mafc76b54.apps.googleusercontent.com';
export const googleClientIdDev = '276806659709-6lap8v4ekmsqqrdaosb3tmiq6j24fvgv.apps.googleusercontent.com';
export const appleTeamId = 'HJ6Q44MPBD';
export const appleClientIdDev = 'TripPicsWebsiteDev';
export const appleClientIdStaging = 'TripPicsWebsiteStaging';
export const appleClientIdProd = 'TripPicsWebsiteProd';
export const appleKeyIdDev = 'CWS2D262UY';
export const appleKeyIdStaging = 'CUBN8UAVWR';
export const appleKeyIdProd = '6U6N9CLL8X';
export const applePrivateKeyParamName = '/oauth/apple/key-secret';
export const refreshTokenValidityInDays = 365;
export const idTokenValidityInDays = 1; // max is 1 day
export const lambdaPayloadLimit = 6 * 1024 * 1024;
export const DDB_BATCHGET_MAX_SIZE = 100;
export const DDB_BATCHWRITE_MAX_SIZE = 25;