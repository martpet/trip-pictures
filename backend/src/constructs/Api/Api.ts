import { HttpApi } from '@aws-cdk/aws-apigatewayv2-alpha';
import { Fn, NestedStack, RemovalPolicy } from 'aws-cdk-lib';
import { Distribution } from 'aws-cdk-lib/aws-cloudfront';
import { HttpOrigin } from 'aws-cdk-lib/aws-cloudfront-origins';
import { ARecord, RecordTarget } from 'aws-cdk-lib/aws-route53';
import { CloudFrontTarget } from 'aws-cdk-lib/aws-route53-targets';
import { Bucket } from 'aws-cdk-lib/aws-s3';
import { Construct } from 'constructs';

import { Auth, Photos, Zone } from '~/constructs';
import { apiSubdomain, appEnvs } from '~/consts';
import { getCdkEnv } from '~/utils';

import { createDistroBehaviors } from './endpoints/createDistroBehaviors';
import { createRoutes } from './endpoints/createRoutes';

type Props = {
  zone: Zone;
  auth: Auth;
  photos: Photos;
};

export class Api extends NestedStack {
  constructor(scope: Construct, { zone, auth, photos }: Props) {
    super(scope, 'Api');

    const envName = getCdkEnv();
    const { appDomain } = appEnvs[envName];
    const apiDomain = `${apiSubdomain}.${appDomain}`;
    const api = new HttpApi(this, 'http-api');
    const origin = new HttpOrigin(Fn.select(1, Fn.split('://', api.apiEndpoint)));

    const logBucket = new Bucket(this, 'Distro-Log-Bucket', {
      removalPolicy: RemovalPolicy.DESTROY,
      autoDeleteObjects: true,
    });

    const distribution = new Distribution(this, 'Distro', {
      defaultBehavior: { origin },
      domainNames: [apiDomain],
      certificate: zone.certificate,
      enableLogging: true,
      logIncludesCookies: true,
      logBucket,
    });

    createRoutes({ scope, api, auth, photos });

    createDistroBehaviors({
      scope,
      api,
      auth,
      distribution,
      origin,
    });

    new ARecord(this, 'A-Record', {
      zone: zone.hostedZone,
      recordName: apiSubdomain,
      target: RecordTarget.fromAlias(new CloudFrontTarget(distribution)),
    });
  }
}
