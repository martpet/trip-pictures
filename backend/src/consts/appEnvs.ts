import { getPersonalDevDomain } from '~/../../shared/utils/getPersonalDevDomain';
import {
  appleClientIdDev,
  appleClientIdProd,
  appleClientIdStaging,
  appleKeyIdDev,
  appleKeyIdProd,
  appleKeyIdStaging,
  devAccountServiceRoleArn,
  devHostedZoneId,
  googleClientIdDev,
  googleClientIdProd,
  googleClientIdStaging,
  healthCheckAlarmEmails,
  prodAccountId,
  region,
  rootDomain,
  rootHostedZoneId,
  stagingAccountId,
  stagingHostedZoneId,
} from '~/consts';
import { AppEnv, EnvName } from '~/types';

export const appEnvs: Record<EnvName, AppEnv> = {
  production: {
    appDomain: rootDomain,
    hostedZoneId: rootHostedZoneId,
    healthCheckAlarmEmails,
    googleClientId: googleClientIdProd,
    appleClientId: appleClientIdProd,
    appleKeyId: appleKeyIdProd,
    env: {
      account: prodAccountId,
      region,
    },
  },
  staging: {
    appDomain: `test.${rootDomain}`,
    hostedZoneId: stagingHostedZoneId,
    healthCheckAlarmEmails,
    googleClientId: googleClientIdStaging,
    appleClientId: appleClientIdStaging,
    appleKeyId: appleKeyIdStaging,
    env: {
      account: stagingAccountId,
      region,
    },
  },
  personal: {
    appDomain: getPersonalDevDomain(process.env),
    parentHostedZoneId: devHostedZoneId,
    parentHostedZoneRoleArn: devAccountServiceRoleArn,
    googleClientId: googleClientIdDev,
    appleClientId: appleClientIdDev,
    appleKeyId: appleKeyIdDev,
    oauthSecretsRoleArn: devAccountServiceRoleArn,
    env: {
      account: process.env.PERSONAL_DEV_ACCOUNT_ID,
      region: process.env.PERSONAL_DEV_ACCOUNT_REGION,
    },
  },
};
