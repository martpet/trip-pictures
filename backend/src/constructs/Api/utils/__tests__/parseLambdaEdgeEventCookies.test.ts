import { lambdaEdgeViewerEvent } from '~/constructs/Api/consts';
import {
  itCalls,
  itReturns,
  parseEventCookies,
  parseLambdaEdgeEventCookies,
} from '~/constructs/Api/utils';

vi.mock('../parseEventCookies');

const args = [lambdaEdgeViewerEvent] as Parameters<typeof parseLambdaEdgeEventCookies>;

describe('parseLambdaEdgeEventCookies', () => {
  itCalls(parseEventCookies, parseLambdaEdgeEventCookies, args);
  itReturns(parseLambdaEdgeEventCookies, args);
});
