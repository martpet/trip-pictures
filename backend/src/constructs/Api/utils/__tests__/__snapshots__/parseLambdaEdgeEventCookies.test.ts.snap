// Vitest Snapshot v1

exports[`parseLambdaEdgeEventCookies > calls "parseEventCookies" with correct args 1`] = `
[
  [
    {
      "cookies": [
        "dummyCookieValue",
      ],
    },
  ],
]
`;

exports[`parseLambdaEdgeEventCookies > returns a correct value 1`] = `
{
  "dummyEventCookie": "dummyEventCookieValue",
}
`;
